apis/project.cattle.io/v3/pipeline_types.go

```diff
@@ -225,6 +227,24 @@ type ApplyYamlConfig struct {
 	Namespace string `json:"namespace,omitempty" yaml:"namespace,omitempty"`
 }
 
+type PublishCatalogConfig struct {
+	Path      string `json:"path,omitempty" yaml:"path,omitempty"`
+	Catalog   string `json:"catalog,omitempty" yaml:"catalog,omitempty"`
+	Version   string `json:"version,omitempty" yaml:"version,omitempty"`
+	GitURL    string `json:"gitUrl,omitempty" yaml:"gitUrl,omitempty"`
+	GitBranch string `json:"gitBranch,omitempty" yaml:"gitBranch,omitempty"`
+	GitAuthor string `json:"gitAuthor,omitempty" yaml:"gitAuthor,omitempty"`
+	GitEmail  string `json:"gitEmail,omitempty" yaml:"gitEmail,omitempty"`
+}
+
+type ApplyAppConfig struct {
+	Catalog         string `json:"catalog,omitempty" yaml:"catalog,omitempty"`
+	Version         string `json:"version,omitempty" yaml:"version,omitempty"`
+	Answers         string `json:"answers,omitempty" yaml:"answers,omitempty"`
```

Comment:
Should answers be map[string]string? is it expected that user has to put whole json blob into answers field?

Reply:
Yes, map makes more sense.